autotest/gdrivers/wmts.py

```diff
@@ -1759,7 +1760,70 @@ def test_wmts_check_no_overflow_zoom_level():
 
     gdal.Unlink(inputXml)
 
+###############################################################################
+# Test when local wmts tiles are missing
+
+
+def test_wmts_24():
+
+    if gdaltest.wmts_drv is None:
+        pytest.skip()
+
+    gdal.FileFromMemBuffer('/vsimem/wmts_missing_local_tiles.xml', """<Capabilities>
+    <Contents>
+        <Layer>
+            <ows:WGS84BoundingBox>
+                <ows:LowerCorner>-6.38153862706 55.6179644952</ows:LowerCorner>
+                <ows:UpperCorner>60.3815386271 75.5825702342</ows:UpperCorner>
+            </ows:WGS84BoundingBox>
+            <Identifier>lyr1</Identifier>
+            <Title>My layer1</Title>
+            <Style isDefault="true">
+                <Identifier>default_style</Identifier>
+                <Title>Default style</Title>
+            </Style>
+            <TileMatrixSetLink>
+                <TileMatrixSet>tms</TileMatrixSet>
+            </TileMatrixSetLink>
+            <ResourceURL format="image/png"
+    template="file:///invalid_path/{Style}/{TileMatrixSet}/{TileMatrix}/{TileRow}/{TileCol}.png" resourceType="tile"/>
+        </Layer>
+        <TileMatrixSet>
+            <Identifier>tms</Identifier>
+            <ows:Identifier>tms</ows:Identifier>
+            <ows:SupportedCRS>urn:ogc:def:crs:EPSG::3067</ows:SupportedCRS>
+            <TileMatrix>
+                <ows:Identifier>13</ows:Identifier>
+                <ScaleDenominator>3571.42857143</ScaleDenominator>
+                <TopLeftCorner>-548576.0 8388608.0</TopLeftCorner>
+                <TileWidth>256</TileWidth>
+                <TileHeight>256</TileHeight>
+                <MatrixWidth>8192</MatrixWidth>
+                <MatrixHeight>8192</MatrixHeight>
+            </TileMatrix>
+        </TileMatrixSet>
+    </Contents>
+    <ServiceMetadataURL xlink:href="/vsimem/wmts_missing_local_tiles.xml"/>
+</Capabilities>""")
+
+    ds = gdal.Open('WMTS:/vsimem/wmts_missing_local_tiles.xml')
+    assert ds is not None
+    assert ds.RasterXSize == 2097152
```

Comment:
those initial checks coming from test_wmts_22() could be removed, as they are redundant and don't actually test any change due to this pull request. Just keep what is below "#Read some data from the image"